apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionNameInstrumentation.java

```diff
@@ -54,18 +57,24 @@
 
     private final Collection<String> applicationPackages;
     private final JaxRsConfiguration configuration;
+    @Nullable
+    @VisibleForAdvice
+    public static JaxRsTransactionHelper jaxRsTransactionHelper;
 
     public JaxRsTransactionNameInstrumentation(ElasticApmTracer tracer) {
         applicationPackages = tracer.getConfig(StacktraceConfiguration.class).getApplicationPackages();
         configuration = tracer.getConfig(JaxRsConfiguration.class);
+        jaxRsTransactionHelper = new JaxRsTransactionHelper(tracer);
     }
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    private static void setTransactionName(@SimpleMethodSignature String signature) {
+    private static void setTransactionName(@SimpleMethodSignature String signature,
+                                           @AnnotationValueOffsetMappingFactory.AnnotationValueExtractor(annotationClassName = "javax.ws.rs.Path", method = "value", type = AnnotationValueOffsetMappingFactory.AnnotationType.CLASS) String pathAnnotationValue) {
```

Comment:
Create a dedicated `OffsetMapping.Factory` for the JAX-RS use-case. If possible, share code with the `OffsetMapping.Factory` used for the `CaptureTransaction` use-case. But sharing the code is only a nice-to-have rather than a requirement if it turns out to add more complexity than benefit. Create a custom `@JaxRsPath` annotation which takes no parameters. Also take into account the `@Path` annotation of superclasses and interfaces.

Reply:
@felixbarny added dedicated JaxRsOffsetMappingFactory